src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseConcreteTypeAnalyzer.Collector.cs

```diff
@@ -0,0 +1,405 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    public partial class UseConcreteTypeAnalyzer
+    {
+        private sealed class Collector
+        {
+            private readonly State _state;
+
+            public Collector(State state)
+            {
+                _state = state;
+            }
+
+            /// <summary>
+            /// Identify fields/locals/params that are used as 'this' for a virtual dispatch.
+            /// </summary>
+            public void HandleInvocation(IInvocationOperation op)
+            {
+                if (op.IsVirtual)
+                {
+                    if (op.Instance != null)
+                    {
+                        var instance = op.Instance;
+                        if (instance.Kind == OperationKind.ConditionalAccessInstance)
+                        {
+                            var parent = op.Parent;
+                            while (parent != null && parent.Kind != OperationKind.ConditionalAccess)
+                            {
+                                parent = parent.Parent;
+                            }
+
+                            if (parent != null)
+                            {
+                                instance = ((IConditionalAccessOperation)parent).Operation;
+                            }
+                        }
```

Comment:
Can we instead use this helper extension method? https://github.com/dotnet/roslyn-analyzers/blob/10632e6dfeb051173788f4e5cd616abeba6d9ff0/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L308-L311

Reply:
Done.